express-notes-api/index.js

```diff
@@ -0,0 +1,90 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+
+app.use(express.json());
+
+// GET
+
+app.get('/api/notes', (req, res) => {
+  const arr = [];
+  const notes = data.notes;
+  for (const property in notes) {
+    arr.push(notes[property]);
+  }
+  res.send(arr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const notes = data.notes;
+  if (isNaN(req.params.id) || req.params.id < 0) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  }
+  if (notes[req.params.id]) {
+    res.status(200).send(notes[req.params.id]);
+  } else {
+    return res.status(404).send({ error: `cannot find note with id ${req.params.id}` });
+  }
+});
+
+// POST
+
+app.post('/api/notes', (req, res) => {
+  if (!req.body.content) {
+    res.status(400).send({ error: 'content is a required field' });
+  }
+  const newNote = { id: data.nextId, ...req.body };
+  data.notes = { ...data.notes, [data.nextId]: newNote };
+  if (data.notes[newNote.id]) {
+    data.nextId++;
+    return res.status(201).send({ content: `${req.body.content}` });
```

Comment:
Before sending a response to the client you need to update the `data.json` file with the updated data